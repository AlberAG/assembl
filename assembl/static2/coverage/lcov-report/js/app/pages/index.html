<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for js/app/pages</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> js/app/pages
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">35.58% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>201/565</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">29.13% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>97/333</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">28.11% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>52/185</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">36.59% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>191/522</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="brightMirrorFiction.jsx"><a href="brightMirrorFiction.jsx.html">brightMirrorFiction.jsx</a></td>
	<td data-value="70" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 70%;"></div><div class="cover-empty" style="width:30%;"></div></div></td>
	<td data-value="70" class="pct medium">70%</td>
	<td data-value="60" class="abs medium">42/60</td>
	<td data-value="69.05" class="pct medium">69.05%</td>
	<td data-value="42" class="abs medium">29/42</td>
	<td data-value="37.5" class="pct low">37.5%</td>
	<td data-value="16" class="abs low">6/16</td>
	<td data-value="70.91" class="pct medium">70.91%</td>
	<td data-value="55" class="abs medium">39/55</td>
	</tr>

<tr>
	<td class="file high" data-value="community.jsx"><a href="community.jsx.html">community.jsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file low" data-value="idea.jsx"><a href="idea.jsx.html">idea.jsx</a></td>
	<td data-value="36.92" class="pic low"><div class="chart"><div class="cover-fill" style="width: 36%;"></div><div class="cover-empty" style="width:64%;"></div></div></td>
	<td data-value="36.92" class="pct low">36.92%</td>
	<td data-value="130" class="abs low">48/130</td>
	<td data-value="31.51" class="pct low">31.51%</td>
	<td data-value="73" class="abs low">23/73</td>
	<td data-value="37.93" class="pct low">37.93%</td>
	<td data-value="29" class="abs low">11/29</td>
	<td data-value="38.1" class="pct low">38.1%</td>
	<td data-value="126" class="abs low">48/126</td>
	</tr>

<tr>
	<td class="file medium" data-value="question.jsx"><a href="question.jsx.html">question.jsx</a></td>
	<td data-value="61.54" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 61%;"></div><div class="cover-empty" style="width:39%;"></div></div></td>
	<td data-value="61.54" class="pct medium">61.54%</td>
	<td data-value="13" class="abs medium">8/13</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="12" class="abs medium">8/12</td>
	</tr>

<tr>
	<td class="file medium" data-value="requestPasswordChange.jsx"><a href="requestPasswordChange.jsx.html">requestPasswordChange.jsx</a></td>
	<td data-value="57.14" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 57%;"></div><div class="cover-empty" style="width:43%;"></div></div></td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="7" class="abs medium">4/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="5" class="abs high">4/5</td>
	</tr>

<tr>
	<td class="file medium" data-value="syntheses.jsx"><a href="syntheses.jsx.html">syntheses.jsx</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="12" class="abs medium">6/12</td>
	<td data-value="37.5" class="pct low">37.5%</td>
	<td data-value="16" class="abs low">6/16</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	<td data-value="54.55" class="pct medium">54.55%</td>
	<td data-value="11" class="abs medium">6/11</td>
	</tr>

<tr>
	<td class="file low" data-value="synthesis.jsx"><a href="synthesis.jsx.html">synthesis.jsx</a></td>
	<td data-value="27.66" class="pic low"><div class="chart"><div class="cover-fill" style="width: 27%;"></div><div class="cover-empty" style="width:73%;"></div></div></td>
	<td data-value="27.66" class="pct low">27.66%</td>
	<td data-value="47" class="abs low">13/47</td>
	<td data-value="9.09" class="pct low">9.09%</td>
	<td data-value="33" class="abs low">3/33</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="14" class="abs low">4/14</td>
	<td data-value="28.26" class="pct low">28.26%</td>
	<td data-value="46" class="abs low">13/46</td>
	</tr>

<tr>
	<td class="file medium" data-value="voteSession.jsx"><a href="voteSession.jsx.html">voteSession.jsx</a></td>
	<td data-value="51.33" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 51%;"></div><div class="cover-empty" style="width:49%;"></div></div></td>
	<td data-value="51.33" class="pct medium">51.33%</td>
	<td data-value="113" class="abs medium">58/113</td>
	<td data-value="36" class="pct low">36%</td>
	<td data-value="75" class="abs low">27/75</td>
	<td data-value="44" class="pct low">44%</td>
	<td data-value="50" class="abs low">22/50</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="106" class="abs medium">53/106</td>
	</tr>

<tr>
	<td class="file low" data-value="voteSessionAdmin.jsx"><a href="voteSessionAdmin.jsx.html">voteSessionAdmin.jsx</a></td>
	<td data-value="11.54" class="pic low"><div class="chart"><div class="cover-fill" style="width: 11%;"></div><div class="cover-empty" style="width:89%;"></div></div></td>
	<td data-value="11.54" class="pct low">11.54%</td>
	<td data-value="182" class="abs low">21/182</td>
	<td data-value="8.14" class="pct low">8.14%</td>
	<td data-value="86" class="abs low">7/86</td>
	<td data-value="4.76" class="pct low">4.76%</td>
	<td data-value="63" class="abs low">3/63</td>
	<td data-value="11.88" class="pct low">11.88%</td>
	<td data-value="160" class="abs low">19/160</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 26 2018 15:51:30 GMT+0200 (CEST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
