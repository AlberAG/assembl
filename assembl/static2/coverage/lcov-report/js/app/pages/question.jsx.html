<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for js/app/pages/question.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">js/app/pages</a> question.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">61.54% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/12</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
import React from 'react';
import { connect } from 'react-redux';
import { compose, graphql } from 'react-apollo';
import { Translate } from 'react-redux-i18n';
import { Grid } from 'react-bootstrap';
import { Link } from 'react-router';
&nbsp;
import manageErrorAndLoading from '../components/common/manageErrorAndLoading';
import Header from '../components/common/header';
import Posts from '../components/debate/survey/posts';
import Question from '../graphql/QuestionQuery.graphql';
import { get as getRoute } from '../utils/routeMap';
import HeaderStatistics, { statContributions, statMessages, statParticipants } from '../components/common/headerStatistics';
import { getIsPhaseCompletedById } from '../utils/timeline';
&nbsp;
type NavigationParams = {
  questionIndex: string,
  questionId: string
};
&nbsp;
type Props = {
  phaseId: string,
  timeline: Timeline,
  title: string,
  numContributors: number,
  numPosts: number,
  params: NavigationParams,
  thematicTitle: string,
  thematicId: string,
  imgUrl: string,
  slug: string,
  totalSentiments: number
};
&nbsp;
export function DumbQuestion(props: Props) {
  const {
    phaseId,
    imgUrl,
    timeline,
    title,
    numContributors,
    numPosts,
    thematicTitle,
    thematicId,
    params,
    slug,
    totalSentiments
  } = props;
  const link = `${getRoute('idea', { slug: slug, phase: 'survey', phaseId: phaseId, themeId: thematicId })}`;
  let statElements = [];
  const numContributions = numPosts + totalSentiments;
  statElements = [statMessages(numPosts), statContributions(numContributions), statParticipants(numContributors)];
  const isPhaseCompleted = getIsPhaseCompletedById(timeline, phaseId);
  return (
    &lt;div className="question"&gt;
      &lt;div className="relative"&gt;
        &lt;Header title={thematicTitle} imgUrl={imgUrl} phaseId={phaseId}&gt;
          &lt;HeaderStatistics statElements={statElements} /&gt;
        &lt;/Header&gt;
        &lt;div className="background-light proposals"&gt;
          &lt;Grid fluid className="background-light"&gt;
            &lt;div className="max-container"&gt;
              &lt;div className="margin-xl"&gt;&amp;nbsp;&lt;/div&gt;
              &lt;Link to={link} className="button-diamond-dark button-submit button-dark"&gt;
                &lt;Translate value="debate.question.backToQuestions" /&gt;
                &lt;span className="button-diamond-dark-back" /&gt;
              &lt;/Link&gt;
              &lt;div className="question-title"&gt;
                &lt;div className="title-hyphen"&gt;&amp;nbsp;&lt;/div&gt;
                &lt;h1 className="dark-title-1"&gt;
                  &lt;Translate value="debate.survey.proposalsTitle" /&gt;
                &lt;/h1&gt;
              &lt;/div&gt;
              &lt;div className="center"&gt;
                &lt;h3 className="collapsed-title"&gt;
                  &lt;span&gt;{`${params.questionIndex}/ ${title}`}&lt;/span&gt;
                &lt;/h3&gt;
                &lt;Posts questionId={params.questionId} themeId={thematicId} isPhaseCompleted={isPhaseCompleted} /&gt;
                &lt;div className="back-btn-container"&gt;
                  &lt;Link to={link} className="button-submit button-dark"&gt;
                    &lt;Translate value="debate.question.backToQuestions" /&gt;
                  &lt;/Link&gt;
                &lt;/div&gt;
              &lt;/div&gt;
              &lt;div className="margin-xl"&gt;&amp;nbsp;&lt;/div&gt;
            &lt;/div&gt;
          &lt;/Grid&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    &lt;/div&gt;
  );
}
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >st</span>ate =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
  lang: state.i18n.locale,
  slug: state.debate.debateData.slug,
  timeline: state.timeline
});
&nbsp;
export default compose(
  connect(mapStateToProps),
  graphql(Question, {
    props: <span class="fstat-no" title="function not covered" >({</span> data }) =&gt; {
<span class="cstat-no" title="statement not covered" >      if (data.error || data.loading) {</span>
<span class="cstat-no" title="statement not covered" >        return {</span>
          error: data.error,
          loading: data.loading
        };
      }
&nbsp;
      const { question: { numContributors, numPosts, totalSentiments, thematic: { img, title, id } } } = <span class="cstat-no" title="statement not covered" >data;</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >      return {</span>
        error: data.error,
        loading: data.loading,
        title: data.question.title,
        imgUrl: img ? img.externalUrl : '',
        numContributors: numContributors,
        numPosts: numPosts,
        thematicTitle: title,
        thematicId: id,
        totalSentiments: totalSentiments
      };
    }
  }),
  manageErrorAndLoading({ color: 'black', displayLoader: true })
)(DumbQuestion);</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 26 2018 15:51:30 GMT+0200 (CEST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
