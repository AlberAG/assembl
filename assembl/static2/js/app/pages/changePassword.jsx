/* eslint no-alert: "off" */

import React from 'react';
import { Grid, Row, Col, FormGroup, FormControl, Button } from 'react-bootstrap';
import { Translate, I18n } from 'react-redux-i18n';
import { getAuthorizationToken } from '../utils/globalFunctions';
import { postChangePassword } from '../services/authenticationService';
import inputHandler from '../utils/inputHandler';
import { get } from '../utils/routeMap';

class ChangePassword extends React.Component {
  constructor(props) {
    super(props);
    this.handleChangePassword = this.handleChangePassword.bind(this);
    this.submitForm = this.submitForm.bind(this);

    this.state = {
      token: getAuthorizationToken()
    };
  }

  handleChangePassword(e) {
    inputHandler(this, e);
  }

  submitForm(e) {
    e.preventDefault();
    const payload = this.state;
    const that = this;
    postChangePassword(payload).then(() => {
      const slug = that.props.params.slug;
      if (slug) {
        const route = `/${get('home', { slug: slug })}`;
        const url = new URL(route, window.location.href);
        window.location = url;
      }
      // Get a slug, reload to the home_view
      // If no slug, go to forbidden page...
    })
    .catch((error) => {
      try {
        const resp = JSON.parse(error);
        // TODO: Use the new Alert system from Phase1 branch
        alert(resp.error.error);
      } catch (exception) {
        // TODO: Use the new Alert system from Phase1 branch
        alert(exception);
      }
    });
  }

  render() {
    return (
      <div>
        <div className="box-title">
          <Translate value="login.changePassword" />
        </div>
        <div className="box">
          <Grid fluid className="login-container">
            <Row className="max-container center">
              <Col xs={12} md={6} className="col-centered">
                <form>
                  <FormGroup className="margin-m">
                    <FormControl type="text" name="password1" required onChange={this.handleChangePassword} />
                  </FormGroup>
                  <FormGroup className="margin-m">
                    <FormControl type="text" name="password2" required onChange={this.handleChangePassword} />
                  </FormGroup>
                  <FormGroup>
                    <Button
                      type="submit"
                      name="change_password"
                      value={I18n.t('login.send')} className="button-submit button-dark"
                      onClick={this.submitForm}
                    >
                      <Translate value="login.send" />
                    </Button>
                  </FormGroup>
                </form>
              </Col>
            </Row>
          </Grid>
        </div>
      </div>
    );
  }
}


export default ChangePassword;