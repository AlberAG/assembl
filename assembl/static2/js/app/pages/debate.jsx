import React from 'react';
import { connect } from 'react-redux';
import { isDateExpired } from '../utils/globalFunctions';
import Survey from '../components/debate/survey';
import Thread from '../components/debate/thread';
import TwoColumns from '../components/debate/twoColumns';
import TokenVote from '../components/debate/tokenVote';

class Debate extends React.Component {
  getCurrentStepIdentifier() {
    const currentDate = new Date();
    const { debateData } = this.props.debate;
    let identifier = null;
    debateData.timeline.map((step) => {
      const startDate = new Date(step.start);
      const endDate = new Date(step.end);
      if (isDateExpired(currentDate, startDate) && isDateExpired(endDate, currentDate)) {
        identifier = step.identifier;
      }
      return identifier;
    });
    return identifier || 'Thread';
  }
  render() {
    const currentIdentifier = this.getCurrentStepIdentifier();
    return (
      <div className="debate">
        {currentIdentifier === 'Survey' &&
          <Survey />
        }
        {currentIdentifier === 'Thread' &&
          <Thread />
        }
        {currentIdentifier === 'TwoColumns' &&
          <TwoColumns />
        }
        {currentIdentifier === 'TokenVote' &&
          <TokenVote />
        }
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    debate: state.debate
  };
};

export default connect(mapStateToProps)(Debate);