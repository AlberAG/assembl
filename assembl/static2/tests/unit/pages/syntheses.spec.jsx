import React from 'react';
import renderer from 'react-test-renderer';

import { DumbSyntheses } from '../../../js/app/pages/syntheses';

describe('Syntheses component', () => {
  it('should match Syntheses snapshot', () => {
    const props = {
      syntheses: [
        {
          id: 'fooId',
          subject: 'Foo',
          img: { externalUrl: 'http://foo.com/bar' },
          creationDate: '2017-02-10T09:15:20.707854+00:00'
        },
        {
          id: 'barId',
          subject: 'Bar',
          img: { externalUrl: 'http://foo.com/bar' },
          creationDate: '2017-02-10T09:15:20.707854+00:00'
        }
      ],
      slug: 'fooslug'
    };
    const rendered = renderer.create(<DumbSyntheses {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});