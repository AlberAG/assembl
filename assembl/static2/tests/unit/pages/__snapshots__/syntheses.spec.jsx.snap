// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Syntheses component should match Syntheses snapshot 1`] = `
<section
  className="themes-section"
>
  <div
    className="background-grey container-fluid"
  >
    <div
      className="max-container"
    >
      <div
        className="title-section"
      >
        <div>
          <div
            className="title-hyphen"
          >
             
          </div>
          <h1
            className="section-title section-title1 dark-title-1"
          >
            debate.syntheses
          </h1>
        </div>
      </div>
      <div
        className="content-section"
      >
        <div
          className="no-margin row"
        >
          <div
            className="theme clear theme-first-row-0 col-md-3 col-sm-6 col-xs-12"
          >
            <div
              className="illustration-box synthesis-preview"
            >
              <div
                className="image-box"
                style={
                  Object {
                    "backgroundImage": "url(http://foo.com/bar)",
                  }
                }
              />
              <a
                className="content-box"
                onClick={[Function]}
                style={Object {}}
              >
                <div
                  className="title-container center"
                >
                  <h3
                    className="light-title-3"
                  >
                    Foo
                  </h3>
                  <h4
                    className="light-title-4"
                  >
                    <span
                      className={undefined}
                      style={undefined}
                    >
                      For15amt2
                    </span>
                  </h4>
                </div>
              </a>
              <div
                className="color-box"
              >
                 
              </div>
              <div
                className="box-hyphen"
              >
                 
              </div>
              <div
                className="box-hyphen rotate-hyphen"
              >
                 
              </div>
            </div>
          </div>
          <div
            className="theme theme-first-row-1 col-md-3 col-sm-6 col-xs-12"
          >
            <div
              className="illustration-box synthesis-preview"
            >
              <div
                className="image-box"
                style={
                  Object {
                    "backgroundImage": "url(http://foo.com/bar)",
                  }
                }
              />
              <a
                className="content-box"
                onClick={[Function]}
                style={Object {}}
              >
                <div
                  className="title-container center"
                >
                  <h3
                    className="light-title-3"
                  >
                    Bar
                  </h3>
                  <h4
                    className="light-title-4"
                  >
                    <span
                      className={undefined}
                      style={undefined}
                    >
                      For15amt2
                    </span>
                  </h4>
                </div>
              </a>
              <div
                className="color-box"
              >
                 
              </div>
              <div
                className="box-hyphen"
              >
                 
              </div>
              <div
                className="box-hyphen rotate-hyphen"
              >
                 
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
